driver-core/src/main/java/com/datastax/driver/core/ErrorTracker.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+/**
+ * Interface for objects that are interested in monitoring precisely errors returned by all hosts.
+ *
+ * An implementation of this interface can be registered against a Cluster
+ * object trough the {@link Cluster#register} method, after which the
+ * {@link #notifyError(Host, Statement, Exception)} method will be called after each query of the driver to a Cassandra
+ * host with the error that has been returned for this operation.
+ */
+public interface ErrorTracker {
+
+    /**
+     * Method that is called after each request returning an error code from Cassandra.
+     *
+     * @param host the host that has been queried.
+     * @param statement the statement that has been sent.
+     * @param exception the exception the query caused.
+     */
+    public void notifyError(Host host, Statement statement, Exception exception);
```

Comment:
My thought was that `ErrorTracker`'s functionality is essentially a subset of `LatencyTracker` with the difference being the method name (`update` -> `notifyError`) and that `ErrorTracker` is only called for a subset of events (errors only).

One thought I just had is that `ErrorTracker` could just be an implementation of `LatencyTracker` that filters out non exceptions in its `update` method and calls notifyError, i.e.:

```java
package com.datastax.driver.core;

public abstract class ErrorTracker implements LatencyTracker {
    @Override
    public void update(Host host, Statement statement, Exception exception, long newLatencyNanos) {
        if (exception != null) {
            notifyError(host, statement, exception);
        }
    }

    public abstract void notifyError(Host host, Statement statement, Exception exception);
}
```

The benefit of this is now you no longer need separate logic in core code distinguishing `ErrorTracker` from `LatencyTracker`, so you don't need more overloaded methods, separate variables for trackers, etc.

The downside of having `ErrorTracker` a `LatencyTracker`, is maybe that this could be considered a leaky abstraction and a detail the user maybe doesn't need to be concerned about.  Not saying it's the best solution, just brainstorming :).